model/Hierarchy.php

```diff
@@ -299,12 +330,20 @@ public function markingFilterMatches($node) {
 	 * @param mixed      $context
 	 * @param string     $childrenMethod    The name of the instance method to call to get the object's list of children
 	 * @param string     $numChildrenMethod The name of the instance method to call to count the object's children
+	 * @param int        $nodeCountThreshold
+	 * @param array      $filteredIds list of ids that will be used to filter list items (reduce the result set before processing)
 	 * @return DataList
 	 */
 	public function markChildren($node, $context = null, $childrenMethod = "AllChildrenIncludingDeleted",
-			$numChildrenMethod = "numChildren") {
+			$numChildrenMethod = "numChildren", $nodeCountThreshold = null, array $filteredIds = array()) {
```

Comment:
Would it make sense to move the nodeCountThreshold to a config instead of passing it around? Would also save a bunch of null's being passed around